omod/src/main/java/org/openmrs/module/datafilter/web/UserFormSubmissionHandler.java

```diff
@@ -74,21 +74,19 @@ public void handle(ServletRequest request, ServletResponse response, FilterChain
 				log.info("User details missing or Invalid user details");
 				return;
 			}
-			String[] locationStrings = request.getParameterValues("locationStrings");
-			if (locationStrings == null) {
+
+			String[] locationStrings = request.getParameterValues("locationStrings") != null ? request.getParameterValues("locationStrings") : new String[0];
+			Collection<EntityBasisMap> mappedLocationsMap = dataFilterService.getEntityBasisMaps(user, Location.class.getName());
+			List<OpenmrsObject> locationsToBeRevoked = getLocationsToBeRevoked(locationService.getAllLocations(), mappedLocationsMap, locationStrings);
+			if (!locationsToBeRevoked.isEmpty()) {
+				dataFilterService.revokeAccess(user, locationsToBeRevoked);
+			}
+			if (locationStrings.length < 1) {
 				log.info("No locations selected");
 				return;
 			}
 			Set<OpenmrsObject> allSelectedLocations = Arrays.stream(locationStrings).map(l -> locationService.getLocation(l))
-			        .filter(Objects::nonNull).collect(Collectors.toSet());
-			
-			Collection<EntityBasisMap> mappedLocationsMap = dataFilterService.getEntityBasisMaps(user,
-			    Location.class.getName());
-			List<OpenmrsObject> locationsToBeRevoked = getLocationsToBeRevoked(locationService.getAllLocations(),
-			    mappedLocationsMap, locationStrings);
-			if (!locationsToBeRevoked.isEmpty()) {
-				dataFilterService.revokeAccess(user, locationsToBeRevoked);
-			}
+					.filter(Objects::nonNull).collect(Collectors.toSet());
```

Comment:
we can also have our filter at the same line as the original one not so far as here

Reply:
Hi,
Now it should be done